### PR TITLE
Add function insert_field

### PR DESCRIPTION
**Pull request by sizumita**
### Summary

User can insert field in fields on embed

### Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by SnowyLuma** [comment]
This method already exists as [insert_field_at](https://discordpy.readthedocs.io/en/rewrite/api.html#discord.Embed.set_field_at).

File: discord/embeds.py [hunk 1]
```diff
@@ -483,6 +483,44 @@ def set_field_at(self, index, *, name, value, inline=True):
         field['inline'] = inline
         return self
 
+    def insert_field(self, index, *, name, value, inline=True):
```
Comment (Rapptz) [inline discord/embeds.py]:
```suggestion
    def insert_field_at(self, index, *, name, value, inline=True):
```

File: discord/embeds.py [hunk 2]
```diff
@@ -483,6 +483,44 @@ def set_field_at(self, index, *, name, value, inline=True):
...
+        """Insert a field to the embed object before index.
```
Comment (Rapptz) [inline discord/embeds.py]:
```suggestion
        """Insert a field to the embed object at the index given.
```

File: discord/embeds.py [hunk 3]
```diff
@@ -483,6 +483,44 @@ def set_field_at(self, index, *, name, value, inline=True):
...
+
+        The index must point to a valid pre-existing field.
```
Comment (Rapptz) [inline discord/embeds.py]:
```suggestion
```

It doesn't, but going out-of-bounds in `list.insert` will just insert at the end so this should be documented.

File: discord/embeds.py [hunk 4]
```diff
@@ -483,6 +483,44 @@ def set_field_at(self, index, *, name, value, inline=True):
...
+            The name of the field.
+        value: :class:`str`
+            The value of the field.
+        inline: :class:`bool`
+            Whether the field should be displayed inline.
+
+        Raises
+        -------
+        IndexError
+            An invalid index was provided.
```
Comment (Rapptz) [inline discord/embeds.py]: I'm not exactly aware of a situation where this is raised.

**Pull request closed without merging**